tests/robottelo/test_func_locker.py

```diff
@@ -0,0 +1,321 @@
+        for pid, content in results:
+            self.assertEqual(pid, int(content))
+        # assert that the sum in counter file is the sum of indexes
+        self.assertEqual(int(_read_counter_file()), sum(indexes))
+
+    def test_recursive_lock_function(self):
+        """Ensure that recursive calls to locked function is detected using
+        lock_function decorator"""
+        res = self.pool.apply_async(simple_recursive_lock_function, ())
+        with self.assertRaises(FunctionLockerError) as context:
+            try:
+                res.get(timeout=5)
+            except multiprocessing.TimeoutError:
+                self.fail('function lock recursion not detected')
+
+        self.assertIn('recursion detected', str(context.exception))
+
+    def test_recursive_locking_function(self):
+        """Ensure that recursive calls to locked function is detected using
+        decorator and with statement"""
+        res = self.pool.apply_async(simple_recursive_locking_function, ())
+        with self.assertRaises(FunctionLockerError) as context:
+            try:
+                res.get(timeout=5)
+            except multiprocessing.TimeoutError:
+                self.fail('function lock recursion not detected')
+
+        self.assertIn('recursion detected', str(context.exception))
+
+    def test_recursive_combined_function(self):
+        """Ensure that recursive calls to locked function is detected using
+        decorator and with statement"""
+        res = self.pool.apply_async(simple_recursive_combined_function, ())
+        with self.assertRaises(FunctionLockerError) as context:
+            try:
+                res.get(timeout=5)
+            except multiprocessing.TimeoutError:
+                self.fail('function lock recursion not detected')
+
+        self.assertIn('recursion detected', str(context.exception))
+
+    def test_scoped_lock_function(self):
+        """ ensure that the lock file was created at the right location"""
+        # call the function
+        simple_scoped_lock_function()
+        lock_file_path = os.path.join(
+            tempfile.gettempdir(),
+            TEMP_ROOT_DIR,
+            TEMP_FUNC_LOCK_DIR,
+            NAMESPACE_SCOPE_TEST,
+            SCOPE,
+            '{}.simple_scoped_lock_function.lock'.format(
+                _this_module_name_string)
+        )
+        self.assertTrue(os.path.exists(lock_file_path))
+
+    def test_scoped_with_locking(self):
+        simple_scoped_locking_function()
+        lock_file_path = os.path.join(
+            tempfile.gettempdir(),
+            TEMP_ROOT_DIR,
+            TEMP_FUNC_LOCK_DIR,
+            NAMESPACE_SCOPE_TEST_2,
+            SCOPE_2,
+            '{}.simple_scoped_locking_function.lock'.format(
+                _this_module_name_string)
+        )
+        self.assertTrue(os.path.exists(lock_file_path))
+
+
+@lock_function
+def simple_locked_function(index=None):
```

Comment:
nitpick: i'd rather put these functions before tests for readability purposes (got stuck in a test for around 30 sec trying to understand what's that `simple_locked_function` which is neither defined nor imported 😄 )